src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,465 @@
+    {
+        private const string AccuracySummary = nameof(MLPerfTrainingExecutor.AccuracySummary);
+        private const string PerformanceSummary = nameof(MLPerfTrainingExecutor.PerformanceSummary);
+
+        private IStateManager stateManager;
+        private ISystemManagement systemManager;
+
+        /// <summary>
+        /// Constructor for <see cref="MLPerfTrainingExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public MLPerfTrainingExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+             : base(dependencies, parameters)
+        {
+            this.systemManager = this.Dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManager.StateManager;
+
+            if (string.IsNullOrEmpty(this.DiskFilter))
+            {
+                this.DiskFilter = "SizeGreaterThan:8000gb&OSDisk:false";
+            }
+        }
+
+        /// <summary>
+        /// Disk filter string to filter disks to format.
+        /// </summary>
+        public string DiskFilter
+        {
+            get
+            {
+                // Change the size -----------------------------------------------------------------------------------------------------------------------------
+                string filter = this.Parameters.GetValue<string>(nameof(MLPerfTrainingExecutor.DiskFilter), "SizeGreaterThan:1000gb");
+                // Enforce filter to remove OS disk.
+                filter = $"{filter}&OSDisk:false";
+                return filter;
+            }
+
+            set
+            {
+                this.Parameters[nameof(MLPerfTrainingExecutor.DiskFilter)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training model name (e.g. bert, rnnt, ssd-mobilenet).
+        /// </summary>
+        public string Model
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Model));
+            }
+        }
+
+        /// <summary>
+        /// The current running user
+        /// </summary>
+        public string Username
+        {
+            get
+            {
+                string username = this.Parameters.GetValue<string>(nameof(MLPerfTrainingExecutor.Username));
+                if (string.IsNullOrWhiteSpace(username))
+                {
+                    username = this.GetCurrentUserName(true);
+                }
+
+                return username;
+            }
+        }
+
+        /// <summary>
+        /// ML specific parameter used to control size of training samples
+        /// </summary>
+        public string BatchSize
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.BatchSize));
```

Comment:
I will suggest if possible to have default values for all parameters which are expected from user so that flow is happy even if user missed any parameter, what do you think?

Reply:
Sure, I will add them too.